daphne_worker/src/durable/aggregate_store.rs

```diff
@@ -57,9 +59,20 @@ impl DurableObject for AggregateStore {
         }
     }
 
-    async fn fetch(&mut self, mut req: Request) -> Result<Response> {
+    async fn fetch(&mut self, req: Request) -> Result<Response> {
         let id_hex = self.state.id().to_string();
-        ensure_garbage_collected!(req, self, id_hex, BINDING_DAP_AGGREGATE_STORE);
+        let ControlFlow::Continue(mut req) = super::run_garbage_collection(
+            req,
+            &self.state,
+            &self.env,
+            self.config.deployment,
+            &mut self.touched,
+            id_hex,
+            BINDING_DAP_AGGREGATE_STORE,
+        )
+        .await? else {
+            return Response::from_json(&());
```

Comment:
Under what conditions does this happen, and why return this particular response? (Similar question below.)

Reply:
This is exactly what the code was doing before, but it was hidden behind the macro. I've rewritten this control flow a bit to make it more readable and added a comment.